### PR TITLE
scripts: Clean up setup scripts

### PR DESCRIPTION
**Pull request by microbuilder**
Adds help menus to the setup scripts, and also makes some changes based on shellcheck analysis.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>

### CODE REVIEW

File: setup-bootstrap.sh [hunk 1]
```diff
@@ -1,22 +1,63 @@
...
-set -e
+# Exit on command failure
+set -o errexit
+
+# Fail on unset variable
+# Use "${VARNAME-}" instead of "$VARNAME" to access unset variable(s)
+set -o nounset
+
+# Enable debug mode if $TRACE is set
+# To enable, run with: "TRACE=1 ./script.sh"
```
Comment (d3zd3z) [inline setup-bootstrap.sh:13]: BTW, if you make this `env TRACE=1 ./script.sh` it will work with any shell.

File: setup-bootstrap.sh [hunk 2]
```diff
@@ -1,22 +1,63 @@
...
+fi
 
 # Generate a keypair for initial device provisioning.  The idea is
 # that this private key and certificate can be programmed into the
 # devices, within a class, at factory time.  This will allow these
 # devices to request certificates later to authenticate themselves.
 
 # For now, write them to a single name
-if [ -f certs/BOOTSTRAP.crt -o -f certs/BOOTSTRAP.key ];
+if [ -f certs/BOOTSTRAP.crt ] || [ -f certs/BOOTSTRAP.key ];
```
Comment (d3zd3z) [inline setup-bootstrap.sh:54]: Any particular reason for this change? We're already relying on bash, and the problems with -o stem from posix defining `[` to only take 4 arguments.

Comment (microbuilder) [reply setup-bootstrap.sh:54]: Just blindly following shellcheck out of habit, but you're right that we are explicitly setting bash here.

File: setup-bootstrap.sh (hunk 1 above)
Comment (microbuilder) [reply setup-bootstrap.sh]: Fixed, thanks

**Review by d3zd3z — COMMENTED** [review]
I guess my only other suggestion would be to separate the changes that are just linter improvements from the ones that add documentation. But really, it isn't large enough for that to really be an issue.

**Review by d3zd3z — APPROVED** [review]

**Pull request merged**